### PR TITLE
NOOS-876/hotfix-venv-caching

### PR DESCRIPTION
**Pull request by pieeri**
**SEMVER Update Type:**
- [ ] Major
- [ ] Minor
- [x] Patch

## Description:

- Expected: CI cache size with `pipenv` (21 MB)
https://app.circleci.com/pipelines/github/noosenergy/noos-circleci-orb/458/workflows/46d164da-bd71-4439-8034-3ca01ee1bef9/jobs/2607

- Currently: CI cache size with `uv` (32 B , almost null)
https://app.circleci.com/pipelines/github/noosenergy/noos-circleci-orb/458/workflows/46d164da-bd71-4439-8034-3ca01ee1bef9/jobs/2611

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -130,3 +130,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [0.2.5] - 2024-08-21
```
Comment (zyassine) [inline CHANGELOG.md:130]: We published this version under `v0.3.0` tag.  Let's update the correct version in the changelog

**Comment by pieeri** [comment]
@zyassine @mthiboust 

Even after removing the cache pruning, the artifact uploaded by CircleCI is 32B ; this seems quite small compared to the cache uploaded for `pipenv` virtual env (21 MB).

Can one of you investigate the actual content of `uv` cache? Is such a cache actually encompassing all the downloaded dependency wheels?

**Comment by mthiboust** [comment]
Interesting, thanks for spotting this. I'll try to check this on Monday

**Comment by mthiboust** [comment]
Maybe this can explain the small cache size? Our dependencies do not need to be built from source, so `uv` prefer to redownload them rather than caching them because it is apparently faster in Github Actions (and maybe CircleCI?) ?

https://docs.astral.sh/uv/reference/cli/#uv-cache-prune
> --ci
> Optimize the cache for persistence in a continuous integration environment, like GitHub Actions.
> 
> By default, uv caches both the wheels that it builds from source and the pre-built wheels that it downloads directly, to enable high-performance package installation. In some scenarios, though, persisting pre-built wheels may be undesirable. For example, in GitHub Actions, it’s faster to omit pre-built wheels from the cache and instead have re-download them on each run. However, it typically is faster to cache wheels that are built from source, since the wheel building process can be expensive, especially for extension modules.
> 
> In --ci mode, uv will prune any pre-built wheels from the cache, but retain any wheels that were built from source.

**Comment by pieeri** [comment]
@mthiboust similar behaviour once the caching command has been removed ; refer to the CI integration tests for this PR.

**Review by mthiboust — COMMENTED** [review]
LGTM

**Pull request merged**